MPDZ.py

```diff
@@ -0,0 +1,471 @@
+    "s": "tcg",
+    "Y": "TAC",
+    "y": "tac",
+    "C": "TGC",
+    "c": "tgc",
+    "W": "TGG",
+    "w": "tgg",
+    "P": "CCC",
+    "p": "ccc",
+    "H": "CAT",
+    "h": "cat",
+    "Q": "GAA",
+    "q": "gaa",
+    "R": "CGA",
+    "r": "cga",
+    "I": "ATT",
+    "i": "att",
+    "M": "ATG",
+    "m": "atg",
+    "T": "ACC",
+    "t": "acc",
+    "N": "AAT",
+    "n": "aat",
+    "K": "AAA",
+    "k": "aaa",
+    "V": "GTT",
+    "v": "gtt",
+    "A": "GCA",
+    "a": "gca",
+    "D": "GAT",
+    "d": "gca",
+    "E": "GAG",
+    "e": "gag",
+    "G": "GGG",
+    "g": "ggg",
+}
+
+
+threel = {
+    "A": "ALA",
+    "R": "ARG",
+    "N": "ASN",
+    "D": "ASP",
+    "V": "VAL",
+    "H": "HIS",
+    "G": "GLY",
+    "Q": "GLN",
+    "E": "GLU",
+    "I": "ILE",
+    "L": "LEU",
+    "K": "LYS",
+    "M": "MET",
+    "P": "PRO",
+    "S": "SER",
+    "Y": "TYR",
+    "T": "THR",
+    "W": "TRP",
+    "F": "PHE",
+    "C": "CYS",
+    "a": "ala",
+    "r": "arg",
+    "n": "asn",
+    "d": "asp",
+    "v": "val",
+    "h": "his",
+    "g": "gly",
+    "q": "gln",
+    "e": "glu",
+    "i": "ile",
+    "l": "leu",
+    "k": "lys",
+    "m": "met",
+    "p": "pro",
+    "s": "ser",
+    "y": "tyr",
+    "t": "thr",
+    "w": "trp",
+    "f": "phe",
+    "c": "cys",
+}
+
+
+def protein_tool(*proteins, options=None):
+    proteins = list(proteins)
+
+    operations = {
+        "compare": compare,
+        "length": length_info,
+        "percentage": count_percentage_aa,
+        "pattern": find_pattern,
+        "3Letter_name": rename_three_letter_name,
+        "DNA_code": get_protein_gene,
+    }
+
+    if options == "compare":
+        result = operations[options](proteins[:-2], proteins[-2], proteins[-1])
+        return result
+    elif options == "3Letter_name":
+        result = operations[options](proteins[:-1], proteins[-1])
+        return result
+    elif options == "length" or options == "percentage" or options == "DNA_code":
+        result = []
+        for protein in proteins:
+            res = operations[options](protein)
+            result.append(res)
+        return result
+    else:
+        raise ValueError("Incorrect options input, please try again")
+
+
+from Bio import SeqIO
+#from Bio.SeqUtils import GC
+
+class FastQFilter:
+    def __init__(self, input_file, output_file, min_length, min_quality, min_gc):
+        self.input_file = input_file
+        self.output_file = output_file
+        self.min_length = min_length
+        self.min_quality = min_quality
+        self.min_gc = min_gc
+
+    def filter_fastq(self):
+        with open(self.output_file, 'w') as output_handle:
+            for record in SeqIO.parse(self.input_file, 'fastq'):
+                if (
+                    len(record.seq) >= self.min_length
+                    and min(record.letter_annotations["phred_quality"]) >= self.min_quality
+                    and Bio.SeqUtils.GC(record.seq) >= self.min_gc
+                ):

```

Comment:
Очень лаконичный код, возьму на заметку.
Только, опять же, длина и GC-состав задаются двумя границами:

```suggestion
                if (
                    length_max >= len(record.seq) >= length_min
                    and min(record.letter_annotations["phred_quality"]) >= self.quality
                    and max_gc_content >= GC(record.seq) >= min_gc_content
                ):
```